binding_test.go

```diff
@@ -0,0 +1,388 @@
+	runTests(t, dsn, func(dbt *DBTest) {
+		dbt.mustExec(enableArrow)
+		var err error
+		rows := dbt.mustQuery(
+			"SELECT 1.0::NUMBER(30,2) as C1, 2::NUMBER(38,0) AS C2, 't3' AS C3, 4.2::DOUBLE AS C4, 'abcd'::BINARY AS C5, true AS C6")
+		defer rows.Close()
+		if !rows.Next() {
+			dbt.Error("failed to query")
+		}
+		var v1, v2, v3, v4, v5, v6 interface{}
+		err = rows.Scan(&v1, &v2, &v3, &v4, &v5, &v6)
+		if err != nil {
+			dbt.Errorf("failed to scan: %#v", err)
+		}
+		var s1 *big.Float
+		var s2 int64
+		var s3 string
+		var s4 float64
+		var ok bool
+		s1, ok = v1.(*big.Float)
+		if !ok || s1.Cmp(big.NewFloat(1.0)) != 0 {
+			dbt.Fatalf("failed to fetch. ok: %v, value: %v", ok, v1)
+		}
+		s2, ok = v2.(int64)
+		if !ok || s2 != 2 {
+			dbt.Fatalf("failed to fetch. ok: %v, value: %v", ok, v2)
+		}
+		s3, ok = v3.(string)
+		if !ok || s3 != "t3" {
+			dbt.Fatalf("failed to fetch. ok: %v, value: %v", ok, v3)
+		}
+		s4, ok = v4.(float64)
+		if !ok || s4 != 4.2 {
+			dbt.Fatalf("failed to fetch. ok: %v, value: %v", ok, v4)
+		}
+	})
+}
+
+// TestBindingArray tests basic array binding via the usage of the Array
+// function that converts the passed Golang slice to a Snowflake array type
+func TestBindingArray(t *testing.T) {
+	runTests(t, dsn, func(dbt *DBTest) {
+		dbt.mustExec(createTableSQL)
+		defer dbt.mustExec(deleteTableSQL)
+
+		intArray := []int{1, 2, 3}
+		fltArray := []float64{0.1, 2.34, 5.678}
+		boolArray := []bool{true, false, true}
+		strArray := []string{"test1", "test2", "test3"}
+		dbt.mustExec(insertSQL, Array(&intArray), Array(&fltArray), Array(&boolArray), Array(&strArray))
+		rows := dbt.mustQuery(selectAllSQL)
+		defer rows.Close()
+
+		var v1 int
+		var v2 float64
+		var v3 bool
+		var v4 string
+		if rows.Next() {
+			err := rows.Scan(&v1, &v2, &v3, &v4)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if v1 != 1 && v2 != 0.1 && v3 != true && v4 != "test1" {
+				t.Fatalf("failed to fetch. expected: 1, 0.1, true, test1. got: %v, %v, %v, %v", v1, v2, v3, v4)
+			}
+		} else {
+			t.Error("failed to query")
+		}
+
+		if rows.Next() {
+			err := rows.Scan(&v1, &v2, &v3, &v4)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if v1 != 2 && v2 != 2.34 && v3 != false && v4 != "test2" {
+				t.Fatalf("failed to fetch. expected: 2, 2.34, false, test2. got: %v, %v, %v, %v", v1, v2, v3, v4)
+			}
+		} else {
+			t.Error("failed to query")
+		}
+
+		if rows.Next() {
+			err := rows.Scan(&v1, &v2, &v3, &v4)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if v1 != 3 && v2 != 5.678 && v3 != true && v4 != "test3" {
+				t.Fatalf("failed to fetch. expected: 3, test3. got: %v, %v, %v, %v", v1, v2, v3, v4)
+			}
+		} else {
+			t.Error("failed to query")
+		}
+	})
+}
+
+// TestBindingBulkArray tests array binding via streaming PUT using the staging
+// process
+func TestBindingBulkArray(t *testing.T) {
+	runTests(t, dsn, func(dbt *DBTest) {
+		dbt.mustExec("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1")
```

Comment:
This is not feasible for public user account. Unless you run this tests internally, can you use the actual `CLIENT_STAGE_ARRAY_BINDING_THRESHOLD` value?